test/test_client.py

```diff
@@ -775,14 +777,8 @@ def test_close(self):
         coll = self.client.pymongo_test.bar
 
         self.client.close()
-        self.client.close()
-
-        coll.count_documents({})
-
-        self.client.close()
-        self.client.close()
-
-        coll.count_documents({})
+        self.assertRaises(InvalidOperation, coll.count_documents, {})
+        self.client = rs_or_single_client()
```

Comment:
Here and in the other tests let's create and close a _new_ client instead of closing `self.client`. Closing self.client can be problematic when there are already other objects that reference it (like self.db).

Reply:
Done.